### PR TITLE
page traversal fix, inherit from list, re-reoxygenize

### PR DESCRIPTION
**Pull request by jennybc**
The most noticeable change is making `extract_link()` robust to the complete absence of links in the header. I ran into this when requesting exactly one object, but specifying `.limit = Inf`, because I almost always leave that in there now.

I also made the `gh_response` object inherit from `list`. For completely unrelated reasons I tried to turn `gh()` output back into JSON and `jsonlite::toJSON()` will not operate on it until `list` is one of the classes. This change seems pretty harmless or maybe even positive, yes?

I re-roxygenized.

### CODE REVIEW

**Comment by gaborcsardi** [comment]
Do you mind undoing the formatting changes? Thanks.

**Comment by jennybc** [comment]
I removed the formatting changes cause by comment-reflow and got rid of the `else{}` statement.

File: man/gh.Rd [hunk 1]
```diff
@@ -29,17 +29,18 @@ API calls are used to get them, and this can take a potentially
...
+to \code{"application/vnd.github.v3+json"} and the
+\code{User-Agent} field defaults to
+\code{"https://github.com/gaborcsardi/whoami"}. This can be used
+to, e.g., provide a custom media type, in order to access a
+preview feature of the API.}
 }
 \value{
-Answer from the API.
+Answer from the API as a \code{gh_response} object, which is also a
+  \code{list}.
```
Comment (gaborcsardi) [inline man/gh.Rd]: Could you please remove this one, too? Seems just re-indenting. Sorry for being so picky.

Comment (gaborcsardi) [inline man/gh.Rd]: Oh, wait, this is Rd. Nevermind.

Comment (gaborcsardi) [inline man/gh.Rd]: This is why I don't like generated files in the repo, btw. :)

**Comment by gaborcsardi** [comment]
Thanks much!

**Pull request merged**